src/lib.rs

```diff
@@ -2,3 +2,5 @@
 //! # Typesense
 //!
 //! Welcome to typesense, the rust library for the Typesense API.
+
+mod error;
```

Comment:
I think that this would fix some of the clippy errors
```suggestion
pub mod error;
```

Reply:
I though of doing something like this:

---

```rust
mod error;

pub use error::{Result, TypesenseError};
```

---

Will this be ok?

Reply:
That should work too!